### DISCUSSION TITLE
2024-08-16 금요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#273] [기록보드]
2024-07-26 금요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | |
| 10:00 ~ 11:00 | 기술면접 준비 | |
| 11:00 ~ 13:00 | 집중 개발 time | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 내용

### 트러블 슈팅

- 내용

### 내일 계획

- 내용
```

```
// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>
```

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 프로필 수정 페이지 제작

### 트러블 슈팅

- 프로필 수정 및 삭제 후 버디 프로필 쿼리 무효화가 제대로 동작하지 않는 것 같음. 기다리면 되긴 되는데 너무 느림. 이유가 뭘까?

**Reply by Captain-Kim** [reply]
최종주차 To Do List (김병준)

- [ ] 온도지수 계산 로직 구현
- [ ] 팔로워 리스트 렌더링 페이지 구현
- [ ] 만든 여정 아코디언 -> 페이지로 다시 제작
- [ ] 글쓰기 퍼널 패턴 뒤로가기 시 이전 스텝으로 가게 수정
- [ ] UT 결과보고서 PPT 제작
- [ ] 레디우스 값, 마진 값 손 보기
- [ ] 리치텍스트 에디터 구현
- [ ] 버디 프로필 페이지 chips 간격 4px
- [ ] 추천인기 버디즈 페이지 버디 선정 기준 나타내기 i 아이콘 호버 등
- [ ] 글쓰기 나이설정에서 input에 0을 못 지우는 문제 해결
- [x] 추천인기버디즈 하트로 follow 버튼 추가
- [x] 여정 나가기 기능 구현
- [x] 글쓰기 아이콘 여정 리스트로 옮기기 (아이콘, 색상까지 바꾸기)
- [x] 버디 프로필 수정 페이지 만들기 -> 각 내용 클릭 시 온보딩 페이지 컴포넌트 재사용 했다가 다시 돌아와서 수정 페이지에서 ‘완료’버튼 눌러야 수정되는 flow
- [x] 버디 프로필 컴포넌트 블러 (로그인 세션 없으면)
- [x] 버디 프로필 모바일 헤더 X 아이콘 추가
- [x] 버디 프로필 글씨 크기 바꾸기
- [x] 온도지수 아이콘 넣기
- [x] 글쓰기 펜딩 페이지 이미지 디자이너님께 보내드리기
- [x] 여정 삭제하기 기능 구현 -> 구현 완료 -> 클라이언트에서 로직 수행하라는 팀원 피드백 있어 수정까지 완료
- [x] 글쓰기 나이 설정 레인지 좁히기 / 미니멈 20세로 유효성까지 막고 맥시범은 알아서 낮춰보고 유효성X -> 20~70세로 레인지 범위 수정, 유효성은 20~150세로.
- [x] ‘참여하기’ 클릭 시 인원이 가득찼습니다의 경고가 error -> caution으로 수정 -> TripCard에서 에러 메시지를 라우트에서 직접 받아서 showAlert으로 전달하도록 코드가 작성되어 있는데, caution을 전달할 방법이 생각이 안 나 일단 패스.
- [x] 추천인기버디즈 나이까지 표기하기 -> 온보딩 미등록한 버디가 있어 NaN이 출력되는 경우(반환값이 string이 아닌 경우 렌더링 하지 않도록 수정)
- [x] 버디프로필에서 ‘선호하는 버디를 등록해주세요’ 문구를 ‘~ 없습니다’로 변경
- [x] 온도 지수 -> 버디즈 지수로 바꾸기
- [x] 글쓰기 여정 시작일이 오늘일 때 ‘시작 날짜가 현재 날짜보다 이전입니다’ 경고 출력 문제 해결 -> 사용자에게 착오가 있었던 듯. startDate < now로 조건문이 되어 있어 정상 작동함. -> 아님. 문제가 재현되어서 다시 살펴보니 setStartDateTimestamp(value.start.toString());  이렇게만 담고 있어서 생긴 문제였음. 이렇게 하면 시간 데이터가 담기지 않게 되고 ‘2024-08-12’ 이런 형태로만 담기는데 시간 정보를 포함하지 못해서 이런 경우에는 00:00:00초인 자정을 기준으로 날짜가 생성됨. -> 그렇다 보니 자정에서 1초라도 지나면 과거로 보게 되어 이런 현상이 생김. 따라서 시간은 따로 입력받지 않으니 임의로 23:59:59로 설정하여 어떠한 경우에도 이런 에러가 안 뜨도록 수정함.

**Comment by eunohhh** [comment]
## 개인 기록보드 서식 (오은)

### 처리 할 사항

- [ ] : UI 정돈 전체적으로- 메인 모바일/데스크탑
- [ ] : 로그아웃 해도 팔로우버튼 에러 (나가져도 팔로우 체크하는듯)
- [x] : 나가기 / 대기중 / 참여하기 동작 제대로 이루어지도록 수정
- [x] : isRead 상태에 따라서 보여지기 토글
- [x] : UI 정돈 - 여정카드, 디테일 모바일/데스크탑
- [x] : 이메일 회원가입 비밀번호 강제하기
- [x] : 탑배너 프리페치쿼리
- [x] : 스토리 관련 로직 전체 검토
- [x] : 튜토리얼 반응형 으로 수정
- [x] : 알림 기능 기본
- [x] : 알림 기능 - 참여자가 여러명일때 순차적으로 띄우기 처리
- [x] : 알림 기능 - 참여자 알림 로직 오류 없게 수정
- [x] : 오늘날짜를 여정 시작일로 할 경우 뜨는 에러 처리 - 에러가 안남 / 재현불가

### 처리 된 사항
- [x] : 여정 카드에 플필사진 4개까지 보이게끔
- [x] : 여정 찜하기 
- [x] : 여정 '나기기'표시
- [x] : 글수정 페이지 만들기 모바일
- [x] : 글수정 페이지 만들기 데탑
- [x] : 비번찾기 뒤로가기 동작 수정
- [x] : 스토리 글쓰기 시 옵션 조금 상세하게 
- [x] : 튜토리얼 데스크탑에서 헤더 숨기기 
- [x] : pwa 연구 - 하단 참조(ios에서는 불가능)
- [x] : 스토리 삭제 
- [x] : 스토리 좋아요 
- [x] : 스토리 좋아요 옵티미스틱 
- [x] : 모바일에서 스토리 인풋문제 
- [x] : 홈화면 카드에서도 스토리 좋아요 노출하기
- [x] : 네이버 검수요청 
- [x] : 미들웨어 설정
- [x] : 여정 데스크탑 페이지네이션 
- [x] : 여정 관련 UI 수정작업
- [x] : 여정 모바일 무한스크롤
- [x] : 스토리 데탑/모바일 위치 맞추기
- [x] : 데스크탑 홈에서 마우스 스크롤 동작 수정 - 세로스크롤 가능하게하고, 화살표로 목록 이동하고, 좌우스크롤은 옵션
- [x] : error.js not-found.js 등 없는 기능들 
- [x] : 홈배너 로직 수정 - 탠스택쿼리
- [x] : 메인 스테일타임 점검 - 5분으로 설정
- [x] : 하단바 svgr 및 fixed
- [x] : 프로필 에디트 라우트 이동
- [x] : UT 피드백 처리 
- [x] : 얼러트 모달 수정
- [x] : 여정 리스트 인기순 / 최신순 필터링
- [x] : 여정 리스트 모바일에서 필터 버튼 만들기
- [x] : 여정 카드 hot 로직
- [x] : 여정 카드 국내/국외 로직
- [x] : 스토리 카드에서 하트 삭제
- [x] : 글쓰기페이지 UI 안짤리게
- [x] : 상대플필보기에서 여정보기 시에도 화면 지진현상
- [x] : ai 이미지
- [x] : 글쓰기 페이지 레이아웃 검토
- [x] : 컨디셔널 라우트 - 병준님이 처리하신듯
- [x] : 온보딩으로 보내기처리 - 사인업일때는 쉬움 / 프로바이더일때 처리요망... 완료

### 보류 사항
- [ ] : 회원가입 디바운싱 ?? - 안해
- [ ] : 여정 임시 저장 ?? - 안할래~~~

**Comment by slsl2** [comment]
## 개인 기록보드 서식 (김민지)

### 오늘 한 일
- unread messages 구현

### 트러블 슈팅
- 실시간 반영이 안 되는 문제
- 새 메시지 insert 시에는 반영 됨
- contract 업데이트 시에는 안 되는데 왜인지 모르겠다
=> 임시로 모바일에서는 헤더 뒤로가기 누르면 새로고침 되도록 해 놓음

### To Do List (8/16~ 편집 중)

- [x] : unread messages 계산해서 렌더링
- [x] : unread messages 실시간 렌더링
- [x] : 채팅창 열면 contract의 마지막 읽은 메시지 값 변경
- [x] : 헤더에도 탭메뉴와 동일하게 표시
- [x] : 알림 리스트 페이지 모바일
- [x] : 알림 리스트 페이지 데스크탑
- [x] : 새 알림 표시
- [x] : 알림 리스트 페이지에서 읽으면 unread true 처리
- [x] : unread 변경 시 UI 실시간 반영 <- 은님이 해결해주심
- [x] : 채팅 페이지 데스크탑
- [ ] : ㅇㅇ님이 입장하셨습니다
- [x] : 채팅 입장 시점부터 메시지 뜨도록
- [x] : 채팅 리스트 최근 메시지 순서대로 정렬
- [x] : input UI 손보기
- [x] : n/4 데이터 연결
- [x] : 마지막 메시지 시간 utils 사용해 변경
